client/src/app/tabs/form/FormEditor.js

```diff
@@ -65,6 +62,29 @@ export class FormEditor extends CachedComponent {
       importing: false
     };
 
+    this.engineProfile = new EngineProfileHelper({
+      get: () => {
+        const { form } = this.getCached();
+
+        const schema = form.getSchema();
+
+        return getEngineProfileFromForm(schema, DEFAULT_ENGINE_PROFILE);
+      },
+      set: (engineProfile) => {
+        const { form } = this.getCached();
+
+        const root = form._state.schema;
+
+        const modeling = form.get('modeling');
+
+        modeling.editFormField(root, engineProfile);
+
+        this.setCached({ engineProfile });
```

Comment:
I believe the caching logic can be moved to the helper, too. It is duplicated across all implementations.

Reply:
Let me have another look.

Reply:
Done (cf. https://github.com/camunda/camunda-modeler/pull/2573/commits/913be8c6d9799c1175fa87f900da5d2870935d47#diff-2202f2da3a0e27c5a60258c0242730fac685169548dfc027103cb72378a34609R34).